packages/utils/crates/utils-types/src/primitives.rs

```diff
@@ -0,0 +1,210 @@
+use ethnum::{u256, AsU256};
+use std::ops::{Add, Sub};
+
+pub trait BaseBalance {
+    const SYMBOL: &'static str;
+
+    fn value(&self) -> &u256;
+
+    fn symbol(&self) -> &str {
+        Self::SYMBOL
+    }
+
+    fn to_hex(&self) -> String {
```

Comment:
The downside of this approach is that intermediate traits cannot add default implementations of the super traits.
E.g. if we have `trait TypeBase` and `trait BaseBalance: TypeBase`, the `BaseBalance` cannot add default implementations of `TypeBase`. 

A specific example: the `eq` method (from `TypeBase`) cannot have default implementation in `BaseBalance`, although it would be possible to implement it there. It must be implemented by the final struct.